python/TestHarness/TestHarness.py

```diff
@@ -464,143 +524,143 @@ def checkPBSEmulator(self):
-                                # Not sure I am doing this right: I have to change the TEST_DIR to match the temporary cluster_launcher TEST_DIR location, thus violating the tester.specs...
-                                tester.parameters()['test_dir'] = '/'.join(job[2].split('/')[:-1])
-                                outfile = output_file.read()
-                                output_file.close()
-                                output = tester.processResults(tester.specs['moose_dir'], exit_code, self.options, outfile)
-                                self.testOutputAndFinish(tester, exit_code, outfile)
-                                continue
-                            else:
-                                # I ran into this scenario when the cluster went down, but launched/completed my job :)
-                                reason = 'FAILED (NO STDOUT FILE)'
-                                tester.setStatus(reason, tester.bucket_fail)
-                        elif output_value == 'R':
-                            # Job is currently running
-                            reason = 'RUNNING'
-                        elif output_value == 'E':
-                            # Job is exiting
-                            reason = 'EXITING'
-                        elif output_value == 'Q':
-                            # Job is currently queued
-                            reason = 'QUEUED'
-
-                        if reason != '' and tester.getStatus() != tester.bucket_fail:
-                            tester.setStatus(reason, tester.bucket_pbs)
-                            self.handleTestStatus(tester)
+        # Iterate over all jobs found in the batch file
+        for job_id, job_data in self.pbs_data.iteritems():
+            # retrieve dirpath from pbs
+            dirpath = job_data['test_dir']
+
+            # retrieve 'tests' (-i tests) file location from pbs
+            file = os.path.join(job_data['test_dir'], '../', job_data['input_name'])
+
+            # Populate the input_file_name argument so augmentParameters can format the test_name
+            self.options.input_file_name = job_data['input_name']
+
+            # Enter the directory where the test resides
+            saved_cwd = os.getcwd()
+            sys.path.append(os.path.abspath(dirpath))
+            if os.path.exists(dirpath):
+                os.chdir(dirpath)
+            else:
+                print 'Test/s no longer exists!', job_data['test_dir']
+                return
+
+            # Build a Warehouse to hold the MooseObjects
+            warehouse = Warehouse()
+
+            # Build a Parser to parse the objects
+            parser = Parser(self.factory, warehouse)
+
+            # Parse it
+            parser.parse(file)
+
+            # Retrieve the tests from the warehouse
+            testers = warehouse.getAllObjects()
+
+            # launch PBS test
+            self.launchTest(False, file, dirpath, testers, job_data)
```

Comment:
This section is pretty cool. Looks a lot like the `findAndRunTests` routine. I bet we could factor out the common parts. Maybe in another PR.